pages/view/engagement/_id.vue

```diff
@@ -77,6 +77,9 @@
         :index="index"
         :name="con.keyContactName"
         :orgname="con.orgName"
+        :last="getLastEngagement(con._id).subject"
+        :date="getLastEngagement(con._id).date"
+        :number="getLastEngagement(con._id).numParticipants"
```

Comment:
Would it be possible to simplify this? Get the engagement once and populate the fields accordingly?

Reply:
You are totally correct, very inefficient implementation. I will take another look at it.

Reply:
So I improved the efficiency significantly removing a step of the implementation that would take state and clone it. It wasn't necessary but it was required when I was doing changes to some of the data that are no longer required. As to getting it once and populating the fields, we are populating this data with 2 different arrays, specifically Engagement documents with their related child Contacts and Contacts documents with their related child Engagements documents.

In this page at the first layer we get the Engagement + it's child Contacts data to display the Engagement details and a list of related Contacts. For each one of those Contacts, we need to have their related Engagement information to be able to populate the last engagement field of the list component. Right now, our query returns the first and second layers of data i.e. Engagements or Contacts with their related data so in this case we need to make 2 queries to get the data we need, which is; 1 Engagement with all related Contacts + all those same Contacts with all their Engagement information so we can get the most recent.

[There is a way in Mongoose to get nested nested data in a query](https://stackoverflow.com/a/34444982) and I believe that would be more efficient for the client side but I am not sure if it is worth it in this case. What do you think?

Reply:
Not worth the effort IMO. Thanks for the explanation. :+1: Approving now.